src/CMakeLists.txt

```diff
@@ -57,59 +57,109 @@ set_target_properties(zoe PROPERTIES
-	PUBLIC $<INSTALL_INTERFACE:../include>
+	PUBLIC $<INSTALL_INTERFACE:include>
 )
 
 # CURL
 find_package(CURL REQUIRED)
-if(ZOE_BUILD_SHARED_LIBS)	
-	target_link_libraries(zoe PRIVATE ${CURL_LIBRARIES})
-else()
-	target_link_libraries(zoe PUBLIC ${CURL_LIBRARIES})
-endif()
-
-target_include_directories(zoe PRIVATE ${CURL_INCLUDE_DIRS})
+target_link_libraries(zoe PRIVATE CURL::libcurl)
+target_compile_definitions(zoe PRIVATE WITH_LIBCURL)
 
 # OpenSSL
 find_package(OpenSSL)
+
 if(OpenSSL_FOUND)
+	target_link_libraries(zoe PRIVATE OpenSSL::SSL)
 	target_compile_definitions(zoe PRIVATE WITH_OPENSSL)
+endif()
+
+if(ZOE_BUILD_SHARED_LIBS)
+	set(ZOE_PC_NAME "zoe")
+	set(ZOE_CONFLICTS "zoe-static")
+else()
+	set(ZOE_PC_NAME "zoe-static")
+	set(ZOE_CONFLICTS "zoe")
+endif()
+
+if(OpenSSL_FOUND)
+	set(ZOE_REQUIRES "libssl libcurl")
+else()
+	set(ZOE_REQUIRES "libcurl")
+endif()
 
-	if(ZOE_BUILD_SHARED_LIBS)	
-		target_link_libraries(zoe PRIVATE OpenSSL::SSL OpenSSL::Crypto)
+set(ZOE_LIBS "")
+set(ZOE_PRIVATE_LIBS "")
+set(ZOE_LIB_NAME "$<TARGET_FILE_BASE_NAME:zoe>")
+
+if (WIN32)
+    target_link_libraries(zoe PRIVATE ws2_32 crypt32)
+    if(ZOE_BUILD_SHARED_LIBS)
+        set(ZOE_PRIVATE_LIBS "${ZOE_PRIVATE_LIBS} -lws2_32 -lcrypt32")
 	else()
-		target_link_libraries(zoe PUBLIC OpenSSL::SSL OpenSSL::Crypto)
-	endif()
+		set(ZOE_LIBS "${ZOE_LIBS} -lws2_32 -lcrypt32")
+    endif()
 endif()
 
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+find_package(Threads)
+target_link_libraries(zoe PRIVATE Threads::Threads)
 
-if (WIN32 OR _WIN32)
-	if(ZOE_BUILD_SHARED_LIBS)
-		target_link_libraries(zoe PRIVATE Ws2_32.lib Crypt32.lib)
+if(CMAKE_USE_PTHREADS_INIT)
+    if(ZOE_BUILD_SHARED_LIBS)
+        set(ZOE_PRIVATE_LIBS "${ZOE_PRIVATE_LIBS} -pthread")
 	else()
-		target_link_libraries(zoe PUBLIC Ws2_32.lib Crypt32.lib)
-	endif()
+		set(ZOE_LIBS "${ZOE_LIBS} -pthread")
+    endif()
 endif()
 
 install(TARGETS zoe
-        EXPORT zoeConfig
-        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        OBJECTS DESTINATION ${CMAKE_INSTALL_LIBDIR}
+		EXPORT zoeTargets
+		RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+		LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+		ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+		OBJECTS DESTINATION ${CMAKE_INSTALL_LIBDIR}
 )
 
 install(DIRECTORY ../include/zoe DESTINATION include)
 
-install(EXPORT zoeConfig
-    NAMESPACE zoe::
-    DESTINATION share/zoe
+install(EXPORT zoeTargets
+	NAMESPACE zoe::
+	DESTINATION share/zoe
+)
+
+configure_file(
+	"${PROJECT_SOURCE_DIR}/zoe.pc.in"
+	"${CMAKE_CURRENT_BINARY_DIR}/zoe.pc.in.configured"
+  @ONLY
+)
+
+file(GENERATE
+	OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${ZOE_PC_NAME}.pc"
+    INPUT "${CMAKE_CURRENT_BINARY_DIR}/zoe.pc.in.configured"
+)
+
+install(FILES
+	"${CMAKE_CURRENT_BINARY_DIR}/${ZOE_PC_NAME}.pc"
+	DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
```

Comment:
Wait, this is meant to generate different pc file names?
This is not common practice in pkgconfig and should be avoided IMO.